src/Features/Core/Portable/Snippets/SnippetProviders/AbstractIfSnippetProvider.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
+
+            var syntaxContext = document.GetRequiredLanguageService<ISyntaxContextService>().CreateContext(document, semanticModel, position, cancellationToken);
+            return syntaxContext.IsStatementContext || syntaxContext.IsGlobalStatementContext;
+        }
+
+        protected override Task<ImmutableArray<TextChange>> GenerateSnippetTextChangesAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            var snippetTextChange = GenerateSnippetTextChange(document, position);
+            return Task.FromResult(ImmutableArray.Create(snippetTextChange));
+        }
+
+        private static TextChange GenerateSnippetTextChange(Document document, int position)
+        {
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var ifStatement = generator.IfStatement(generator.TrueLiteralExpression(), Array.Empty<SyntaxNode>());
+
+            return new TextChange(TextSpan.FromBounds(position, position), ifStatement.ToFullString());
+        }
+
+        protected override int GetTargetCaretPosition(ISyntaxFactsService syntaxFacts, SyntaxNode caretTarget)
+        {
+            syntaxFacts.GetPartsOfIfStatement(caretTarget, out _, out var statement);
+
+            // Need to get the statement span start and add 1 to insert between the the curly braces
+            return statement.SpanStart + 1;
+        }
+
+        protected override async Task<SyntaxNode> AnnotateNodesToReformatAsync(Document document,
+            SyntaxAnnotation findSnippetAnnotation, SyntaxAnnotation cursorAnnotation, int position, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var snippetExpressionNode = FindAddedSnippetSyntaxNode(root, position, syntaxFacts);
+            if (snippetExpressionNode is null)
+            {
+                return root;
+            }
+
+            var reformatSnippetNode = snippetExpressionNode.WithAdditionalAnnotations(findSnippetAnnotation, cursorAnnotation, Simplifier.Annotation, Formatter.Annotation);
+            return root.ReplaceNode(snippetExpressionNode, reformatSnippetNode);
+        }
+
+        protected override ImmutableArray<SnippetPlaceholder> GetPlaceHolderLocationsList(SyntaxNode node, ISyntaxFacts syntaxFacts, CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<SnippetPlaceholder>.GetInstance(out var arrayBuilder);
+            syntaxFacts.GetPartsOfIfStatement(node, out var condition, out var _);
+
+            arrayBuilder.Add(new SnippetPlaceholder(identifier: condition.ToString(), placeholderPositions: ImmutableArray.Create(condition.SpanStart)));
```

Comment:
I don't think that should be a problem in this PR specifically, since I'm always inserting "true" in the conditional. I can add a contract that says if my string already contains a $, then I bail?

Reply:
contract call seems reasonable.  just be careful with future PRs/snippets.  We would also have a problem if this becomes genrally extensible.

Reply:
how does hte LSP spec say to handle this? what escaping mechanism do they use?